### PR TITLE
Fix issue in removing items with LayoutAnimation on Android. When LayoutAnimation is enabled, indicesToRemove are incorrect, which makes adjacent views disappear.

### PR DESCRIPTION
**Pull request by vinceyuan**
Fix issue #11828 Removing items with LayoutAnimation causes adjacent views/texts to disappear on Android

Screenshot

I explained my solution in https://github.com/facebook/react-native/issues/11828#issuecomment-273434006

### CODE REVIEW

**Comment by vinceyuan** [comment]
@janicduplessis Please take a look. Thanks.

**Comment by vinceyuan** [comment]
Ping @janicduplessis

**Comment by janicduplessis** [comment]
Thanks for the PR! I'll have a look this week ;)

**Comment by janicduplessis** [comment]
Could you provide a test plan that describe how you tested this?

Also are you sure that indicesToRemove and tagsToDelete are always the same? I tried fixing the indices a while back but ended up finding some edge cases where it didn't work. https://github.com/facebook/react-native/pull/7959

**Comment by vinceyuan** [comment]
@janicduplessis 
I explained my solution in https://github.com/facebook/react-native/issues/11828#issuecomment-273434006
Test plan:
Tested that this fixed the bug in [https://github.com/vinceyuan/ReactNativeOpacityBugDemo](https://github.com/vinceyuan/ReactNativeOpacityBugDemo). See screenshot with this fix:
reactnativeopacitybugfixed

I am not sure that indicesToRemove and tagsToDelete are always the same. Probably they are not same when LayoutAnimation is not enabled. If we totally ignore indicesToRemove, there will be problems in removing items without animation. You can see some views are not removed after I tap `Remove without animation` button in the following screenshot:
reactnativeopacitybugignoreindicestoremove

Probably this fix is not perfect. But with this fix, everything works well with/without LayoutAnimation. 😸

**Comment by freakinruben** [comment]
Does this solve #9549 as well?

**Comment by vinceyuan** [comment]
@freakinruben Probably not. This PR fixes an issue in LayoutAnimation on Android. #9549 doesn't seem to be animation related.

**Comment by vinceyuan** [comment]
@janicduplessis Any comment about this PR? Is it good?

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/layoutanimation/LayoutAnimationController.java [hunk 1]
```diff
@@ -65,6 +65,10 @@ public void reset() {
     mShouldAnimateLayout = false;
   }
 
+  public boolean shouldAnimateLayoutChildrenInView(View parentView) {
```
Comment (janicduplessis) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/layoutanimation/LayoutAnimationController.java]: Can you use `shouldAnimateLayout` instead of adding this new method?

**Comment by vinceyuan** [comment]
@janicduplessis Right. shouldAnimateLayoutChildrenInView is not necessary. I have removed it. Thanks!

Let me think about the ideal solution you mentioned.

**Comment by f0rr0** [comment]
@janicduplessis @vinceyuan Is this good to go? Currently, removing items with Layout Animation throws on Android :/

**Comment by vinceyuan** [comment]
@f0rr0 My PR solved the issue. But janicduplessis doesn't think it is good enough. I am trying to find time to work on it.

**Comment by Antoine-C** [comment]
Hi
Any update on this ?
Thank's

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/fix-issue-in-removing-items-with-layoutanimation-on-android-when-layoutanimation

**Comment by hramos** [comment]
@vinceyuan thanks for your contribution! After reviewing the comments here, it looks like this PR is not going to be accepted in its current state. I'll set it back to closed.

**Comment by jpshelley** [comment]
@vinceyuan @janicduplessis @hramos what is keeping this PR from being merged? Still a bit confused by that. Would love to help in anyway I can. This issue has existed on Android for a long time now, and I think it is also in react-native-windows, so learning how to fix it here will help there.

**Comment by hramos** [comment]
@jpshelley the PR has been closed for a while, which is a huge blocker for getting it merged in. If you're interested in making this change, I suggest going back through the thread and understanding why the reviewers pushed back on the change. If I recall correctly, I closed this PR for two reasons: reviewers rejected the changes and the PR laid dormant without activity for months afterwards.
